application/actions/organization_test.go

```diff
@@ -229,3 +231,156 @@ func (as *ActionSuite) Test_OrganizationCreateRemoveUpdate() {
+	userFixtures.Users[0].AdminRole = models.UserAdminRoleSuperAdmin
+	as.NoError(as.DB.Save(&userFixtures.Users[0]))
+
+	userFixtures.Users[1].AdminRole = models.UserAdminRoleSalesAdmin
+	as.NoError(as.DB.Save(&userFixtures.Users[1]))
+
+	userFixtures.Users[2].UserOrganizations[0].Role = models.UserOrganizationRoleAdmin
+	as.NoError(as.DB.Save(&userFixtures.Users[2].UserOrganizations[0]))
+
+	// Test view org1 for each user. users 0,1,2 should succeed, user 3 should fail
+	viewOrg1Payload := fmt.Sprintf(`{"query": "{organization(id: \"%s\"){id}}"}`, org1.UUID.String())
+	listOrgsPayload := `{"query": "{organizations{id}}"}`
+
+	type testCase struct {
+		Name            string
+		Token           string
+		Payload         string
+		ExpectError     bool
+		ExpectSubStrings []string
+		DontExpectSubStrings []string
+	}
+
+	testCases := []testCase{
+		{
+			Name: "View org 1 as user 0 (super admin)",
+			Token: userFixtures.Users[0].Nickname,
+			Payload: viewOrg1Payload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String()},
+		},
+		{
+			Name: "View org 1 as user 1 (sales admin)",
+			Token: userFixtures.Users[1].Nickname,
+			Payload: viewOrg1Payload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String()},
+		},
+		{
+			Name: "View org 1 as user 2 (org admin)",
+			Token: userFixtures.Users[2].Nickname,
+			Payload: viewOrg1Payload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String()},
+		},
+		{
+			Name: "View org 1 as user 3 (normal user)",
+			Token: userFixtures.Users[3].Nickname,
+			Payload: viewOrg1Payload,
+			ExpectError: true,
+		},
+		{
+			Name: "List orgs as user 0 (super admin)",
+			Token: userFixtures.Users[0].Nickname,
+			Payload: listOrgsPayload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String(), org2.UUID.String()},
+		},
+		{
+			Name: "List orgs as user 1 (sales admin)",
+			Token: userFixtures.Users[1].Nickname,
+			Payload: listOrgsPayload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String(), org2.UUID.String()},
+		},
+		{
+			Name: "List orgs as user 2 (org admin)",
+			Token: userFixtures.Users[2].Nickname,
+			Payload: listOrgsPayload,
+			ExpectError: false,
+			ExpectSubStrings: []string{org1.UUID.String()},
+			DontExpectSubStrings: []string{org2.UUID.String()},
+		},
+		{
+			Name: "List orgs as user 3 (normal user)",
+			Token: userFixtures.Users[3].Nickname,
+			Payload: listOrgsPayload,
+			ExpectError: false,
+			DontExpectSubStrings: []string{org1.UUID.String(), org2.UUID.String()},
+		},
+	}
+
+	for _, tc := range testCases {
+		payload := bytes.NewReader([]byte(tc.Payload))
+		req := httptest.NewRequest("POST", "/gql", payload)
+		resp := httptest.NewRecorder()
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", tc.Token))
+		req.Header.Set("content-type", "application/json")
+
+		as.App.ServeHTTP(resp, req)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Error(err)
+		}
```

Comment:
This test is a little different, but just wanted to highlight `actions.testGqlQuery` that does all of this and more.

Reply:
I saw that but didn't look into exactly how it worked and I just used the same process/convention I had for a previous test. should I refactor these to use the shared one?

Reply:
It's up to you. The other method parses the received data into a struct so you can compare field values.

Reply:
Yours is better set up for multiple tests/scenarios, though.